pkgs/by-name/ra/raycast/package.nix

```diff
@@ -3,21 +3,37 @@
   stdenvNoCC,
   fetchurl,
   writeShellApplication,
+  cacert,
   curl,
   jq,
-  common-updater-scripts,
+  openssl,
   undmg,
 }:
 
+let
+  hashes = {
+    aarch64-darwin = "sha256-q3pX/mOl/u9KMcAfvXm4giYKjnTB903N1ibubvaO9Uw=";
+    x86_64-darwin = "sha256-l61AVKx+aYmgnVK8d+by2pKiu1cIAueLipRjOzCvib4=";
+  };
+in
 stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "raycast";
-  version = "1.88.3";
+  version = "1.88.4";
 
-  src = fetchurl {
-    name = "Raycast.dmg";
-    url = "https://releases.raycast.com/releases/${finalAttrs.version}/download?build=universal";
-    hash = "sha256-MOJlEUQHLDw8YeQC9sG5QLSO8qhgcWG8HtyRZCPHb+M=";
-  };
+  src =
+    {
+      aarch64-darwin = fetchurl {
+        name = "Raycast.dmg";
+        url = "https://releases.raycast.com/releases/${finalAttrs.version}/download?build=arm";
+        hash = hashes.${stdenvNoCC.system};
+      };
+      x86_64-darwin = fetchurl {
+        name = "Raycast.dmg";
+        url = "https://releases.raycast.com/releases/${finalAttrs.version}/download?build=x86_64";
+        hash = hashes.${stdenvNoCC.system};
```

Comment:
small nit
is there a reason for not pulling the hashes directly down here?
i strongly prefer the having the hashes right where they are needed instead of in the let-in above 😂

Reply:
I initially kept it at the top since it was easier for me to parse it. I was messing around for the last 20 minutes and came up with this:

https://github.com/NixOS/nixpkgs/blob/625e9b07d80ae695739e4f241679c99058714431/pkgs/by-name/ra/raycast/package.nix#L70-L71

Let me know if there is a more elegant/better way to parse my way to the hashes